crates/nu-command/src/system/benchmark.rs

```diff
@@ -41,12 +45,27 @@ impl Command for Benchmark {
         let redirect_stderr = call.redirect_stderr;
 
         let mut stack = stack.captures_to_stack(&capture_block.captures);
+
+        // In order to provide the pipeline as a positional, it must be converted into a value.
+        // But because pipelines do not have Clone, this one has to be cloned as a value
+        // and then converted back into a pipeline for eval_block().
+        // So, the metadata must be saved here and restored at that point.
+        let input_metadata = input.metadata();
+        let input_val = input.into_value(call.head);
+
+        if let Some(var) = block.signature.get_positional(0) {
+            if let Some(var_id) = &var.var_id {
+                stack.add_var(*var_id, input_val.clone());
+            }
+        }
```

Comment:
Not sure how this affects benchmarking of commands that are streaming regularly?

But in general "caching" the input could also be beneficial for isolated microbenchmarks.